### ISSUE TITLE
sonar:CollapsibleIfStatements doesn't work as expected

### ISSUE THREAD

**Issue by jordiponsllaurado** [walkmod/walkmod-sonar-plugin#4]
This is the code before 'walkmod apply':
```
while (!find) {
    goToTeaserFrame();
    find = browser.elementVisible(By.xpath(xPathTeaserTextField));
     if (!find) {
         if (++retry >= browser.getCustomRetries()) {
             throw new QAException("Unable to access to the text teaser");
        }
    }
}
LOGGER.debug("TextAndImage - Exit: waitUntilTeaserFrameIsAvailable");
```
As you can see there are two consecutive If statements that can be collapsed.

This is the result after 'walkmod apply':
```
while (!find) {
    goToTeaserFrame();
    find = browser.elementVisible(By.xpath(xPathTeaserTextField));
    if (            !find
    && ++retry >= browser.getCustomRetries()) {
        if (++retry >= browser.getCustomRetries()) {
                    }
LOGGER.debug("TextAndImage - Exit: waitUntilTeaserFrameIsAvailable");
```
This result give a compilation error, I think that the result should be something like this:
```
while (!find) {
    goToTeaserFrame();
    find = browser.elementVisible(By.xpath(xPathTeaserTextField));
    if (!find && ++retry >= browser.getCustomRetries()) {
        throw new QAException("Unable to access to the text teaser");
    }
}
LOGGER.debug("TextAndImage - Exit: waitUntilTeaserFrameIsAvailable");
```

My walkmode.xml is this:
```
<!DOCTYPE walkmod PUBLIC "-//WALKMOD//DTD" "http://www.walkmod.com/dtd/walkmod-1.1.dtd">
<walkmod>
    <chain name="default">
        <transformation type="dead-code-cleaner"/>
        <transformation type="sonar:AddCurlyBrackets"/>
        <transformation type="sonar:AddSwitchDefaultCase"/>
        <transformation type="sonar:ArrayDesignatorOnType"/>
        <transformation type="sonar:CollapsibleIfStatements"/>
        <transformation type="sonar:DeclarationsShouldUseCollectionInterfaces"/>
        <transformation type="sonar:PrimitiveInstantiationForToString"/>
        <transformation type="sonar:RedundantCastsShouldNotBeUsed"/>
        <transformation type="sonar:RemoveCodeComment"/>
        <transformation type="sonar:RemoveEmptyMethod"/>
        <transformation type="sonar:RemoveEmptyStatement"/>
        <transformation type="sonar:RemoveLiteralBoolean"/>
        <transformation type="sonar:RemoveUnusedMethodParameters"/>
        <transformation type="sonar:RemoveUselessImports"/>
        <transformation type="sonar:RemoveUselessParentheses"/>
        <transformation type="sonar:RemoveUselessVariables"/>
        <transformation type="sonar:StaticInitializedFieldToFinal"/>
        <transformation type="sonar:StringCheckOnLeft"/>
        <transformation type="sonar:UseCollectionIsEmpty"/>
        <transformation type="sonar:UseStringEquals"/>
        <transformation type="sonar:VariableImmediatelyReturned"/>
        <writer path="src/main/java" type="javalang:string-writer"/>
    </chain>
</walkmod>
```

**Comment by rpau** [comment]
Hi,

The transformation is working correctly. I have created a test and I have exactly what you expect. The problem probably appears when walkmod is reasoning which incremental parts needs to be written in the file.  It is the same than the issue #3 

Probably, if you use the "java-formatter" as a writer, it wouldn't appear. But no worry, let me fix it in few days.

**Comment by rpau** [comment]
Fixed at walkmod-javalang-plugin, commit: 5bcba66e10f913ebd073a659565f68d6eafe74f3
